py-geopolars/python/geopolars/internals/geoseries.py

```diff
@@ -26,6 +29,19 @@ def __init__(self, *args, **kwargs):
 
         super().__init__(*args, **kwargs)
 
+    def __getitem__(
+        self,
+        item: int
+        | pl.Series
+        | GeoSeries
+        | range
+        | slice
+        | np.ndarray[Any, Any]
+        | list[int]
+        | list[bool],
+    ) -> Any:
+        return GeoSeries(super().__getitem__(item))
+
```

Comment:
This means that if you do `GeoSeries(data)[0]` you get a `GeoSeries` back? I think of a `GeoSeries` as representing a collection of geometries, so having it also represent a single scalar seems confusing.

It also seems to be inconsistent with upstream polars where `__getitem__` returns a scalar value.

```py
import polars as pl
s = pl.Series([1, 2, 3, 4])
type(s[0]) # int
```

It's possible we'll need to handle single geometries with a `Geometry` class, but I think that would need more thought for how to implement that in a way that doesn't end up storing separate memory objects for each geometry.

Reply:
Yea. It's a miss in your case when returned a scalar. There was the case when I selected range from geometry column and got back slice of geometry columns but as a series now

What do you thing about returning Geoseries if we were to return Series?

Reply:
Btw what is initial plan to represent geometries in python?
Right now in most cases they are just list of u8. In geopandas shapely is used to represent geom classes like Point and others.

Reply:
Ah I see what you mean. For _nested_ types, the `__getitem__` of a series is another series

```py
import geopolars as gpl
geom_col = gpl.datasets.read_dataset('naturalearth_cities').get_column('geometry')

type(geom_col[0])
# polars.internals.series.series.Series

geom_col[0].dtype
# polars.datatypes.UInt8
```

The issue is that this sub-series no longer fits in with `GeoSeries`; all of the `GeoSeries` methods would fail because `GeoSeries` expects that each element is an entire geometry, not a `u8`.

> Btw what is initial plan to represent geometries in python?

This is hard to say. I think we really want to stick with Arrow memory for geometries, which means that we can't expose geometries directly as Shapely objects. Maybe it would make sense to add geometry class(es) that wrap individual objects, but we'd have to be careful to ensure the memory is a view on existing column memory

As further background: for now in Rust we use store geometries as WKB (with a `List<u8>` array type because Polars doesn't support the `Binary` type directly) but that's only a stopgap measure. We really want to use the [geo-arrow native encoding](https://github.com/geopandas/geo-arrow-spec/blob/main/format.md), so as soon as Polars supports `FixedSizeList` we'll switch over to that, which should make interop with Shapely geometries faster. E.g. Joris benchmarked to/from arrow to be significantly faster than to/from WKB. https://github.com/jorisvandenbossche/python-geoarrow

Reply:
Thanks for clarification. Is there other place for discussion e.g slack, discrod (as polars, or on polars server) ?

So perhaps I'll remove changes, leave as it was and create a separate issue

Reply:
> Thanks for clarification. Is there other place for discussion e.g slack, discrod (as polars, or on polars server) ?

So far we haven't created another place for discussions, so I think issues or [discussions](https://github.com/geopolars/geopolars/discussions) on this repo is the easiest solution

> So perhaps I'll remove changes, leave as it was and create a separate issue

Yeah a new issue would be great for this!